shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -1821,16 +1821,18 @@ private boolean didChangeLabel() {
         }
 
         private void log(@NonNull String indent, boolean recursive) {
-            Log.i(TAG,
-                    indent + "SemanticsNode id=" + id + " label=" + label + " actions=" + actions
-                            + " flags=" + flags + "\n" + indent + "  +-- textDirection="
-                            + textDirection + "\n" + indent + "  +-- rect.ltrb=(" + left + ", "
-                            + top + ", " + right + ", " + bottom + ")\n" + indent
-                            + "  +-- transform=" + Arrays.toString(transform) + "\n");
-            if (childrenInTraversalOrder != null && recursive) {
-                String childIndent = indent + "  ";
-                for (SemanticsNode child : childrenInTraversalOrder) {
-                    child.log(childIndent, recursive);
+            if (BuildConfig.DEBUG) {
+                Log.i(TAG,
```

Comment:
Doesn't Android already take care of avoiding log statements when built in release mode?

Reply:
I don't know if it can because this is a library.  Either way, the linter is telling us we should be guarding this this way.

Reply:
(specifically for logging levels below warning)

Reply:
Is there someone who could clarify whether we need this in actuality? I've got a lot of logs in the embedding and unfortunately this is quite a tedious practice per log statement. I would definitely appreciate if we could determine with confidence what the behavior is and ideally avoid this. Another option might be to introduce our own Log class that internally contains these guards.

Reply:
Only needed for log levels below warn.  Linter clarifies it, right?

Reply:
Plus we don't want to be adding more noise to logcat for informational logging outside of debug mode do we?

Reply:
The documentation says that [verbose logs aren't compiled into release build, and debug logs are but are stripped at runtime](https://developer.android.com/reference/android/util/Log.html). `INFO` statements are higher than both so would always be included, even in regular apps. I'm having trouble finding info on if this automatic stripping is different for libraries vs apps. [This page on releasing your app](https://developer.android.com/studio/publish#publishing-configure) says to manually remove Log calls, so seems like there's conflicting info floating around. I know proguard could handle this too, but we're not using it.

We could do a quick check to confirm that just logs out at every level when the engine initializes and check what gets printed with `flutter run --release`?

+1 to our own Log class wrapper instead of needing to guard at every call site though.

Reply:
I'm not strongly opposed to that, but I think we should generally avoid informational logging in the embedder outside of temporary logs for local debugging.  Creating a utility class could encourage people to add logging that we don't need.

IOW, informational logging that remains in this codebase _should_ be painful. We only have a few callsites (some of which are already in helper classes or methods), and even of those I think two of them should be upgraded to errors or warnings.

Reply:
(I say this in part because logcat is already super chatty, and we should avoid adding to chatter when you're looking for flutter specific logs)

Reply:
Understood. I think requiring the callsite checks makes sense then.